object-manipulation-2/src/pick.js

```diff
@@ -1,14 +1,11 @@
 /* exported pick */
 
 function pick(source, keys) {
-
   var output = {};
-
-  for (var key in source) {
-
-    if (source[key] !== undefined) {
-      for (var i = 0; i < keys.length; i++) {
-        if (key === keys[i]) {
+  for (var i = 0; i < keys.length; i++) {
+    for (var key in source) {
+      if (key === keys[i]) {
+        if (source[key] !== undefined) {
```

Comment:
You can massively simplify this by making your condition `if(source[keys[i]] !== undefined)`